e2e/framework/testserver.go

```diff
@@ -550,6 +563,171 @@ type PreparedTransaction struct {
 	Input    []byte
 }
 
+type Txn struct {
+	key     *wallet.Key
+	client  *jsonrpc.Eth
+	hash    *ethgo.Hash
+	receipt *ethgo.Receipt
+	raw     *ethgo.Transaction
+	chainID *big.Int
+
+	sendErr error
+	waitErr error
+}
+
+func (t *Txn) Deploy(input []byte) *Txn {
+	t.raw.Input = input
+
+	return t
+}
+
+func (t *Txn) Transfer(to ethgo.Address, value *big.Int) *Txn {
+	t.raw.To = &to
+	t.raw.Value = value
+
+	return t
+}
+
+func (t *Txn) Value(value *big.Int) *Txn {
+	t.raw.Value = value
+
+	return t
+}
+
+func (t *Txn) To(to ethgo.Address) *Txn {
+	t.raw.To = &to
+
+	return t
+}
```

Comment:
Nitpick, Are `Value` and `To` methods needed? because `Txn` has `Transfer` already

Reply:
`Transfer` is a sugarcoat version that combines `Value` and `To`. There might be a situation when you want to deploy a contract but also send a value.